packages/framework/tests/Unit/Component/Money/MoneyTest.php

```diff
@@ -0,0 +1,582 @@
+    {
+        yield ['1', '1', '2'];
+        yield ['12.15', '34.965', '47.115'];
+        yield ['10', '-2', '8'];
+        yield ['1', '0.01', '1.01'];
+        yield ['0.5', '0.5', '1.0'];
+        yield ['1.525', '0.475', '2.000'];
+        yield ['1.00', '1.000', '2.000'];
+    }
+
+    public function testSubtractIsImmutable(): void
+    {
+        $money = Money::fromString('1');
+
+        $money->subtract(Money::fromString('1'));
+
+        $this->assertEquals('1', $money->toString());
+    }
+
+    /**
+     * @dataProvider subtractProvider
+     * @param string $a
+     * @param string $b
+     * @param string $expectedResult
+     */
+    public function testSubtract(string $a, string $b, string $expectedResult): void
+    {
+        $moneyA = Money::fromString($a);
+        $moneyB = Money::fromString($b);
+
+        $result = $moneyA->subtract($moneyB);
+
+        $this->assertEquals($expectedResult, $result->toString());
+    }
+
+    /**
+     * @return \Iterator
+     */
+    public function subtractProvider(): Iterator
+    {
+        yield ['2', '1', '1'];
+        yield ['12.15', '34.965', '-22.815'];
+        yield ['10', '-2', '12'];
+        yield ['1', '0.01', '0.99'];
+        yield ['0.5', '0.5', '0.0'];
+        yield ['1.525', '0.475', '1.050'];
+        yield ['1.00', '1.000', '0.000'];
+    }
+
+    public function testMultiplyIsImmutable(): void
+    {
+        $money = Money::fromString('1');
+
+        $money->multiply('2');
+
+        $this->assertEquals('1', $money->toString());
+    }
+
+    /**
+     * @dataProvider multiplyProvider
+     * @param string $a
+     * @param string $b
+     * @param string $expectedResult
+     */
+    public function testMultiply(string $a, string $b, string $expectedResult): void
+    {
+        $moneyA = Money::fromString($a);
+
+        $result = $moneyA->multiply($b);
+
+        $this->assertEquals($expectedResult, $result->toString());
+    }
+
+    /**
+     * @return \Iterator
+     */
+    public function multiplyProvider(): Iterator
+    {
+        yield ['2', '1', '2'];
+        yield ['12.15', '34.965', '424.82475'];
+        yield ['10', '-2', '-20'];
+        yield ['1', '0.01', '0.01'];
+        yield ['0.5', '0.5', '0.25'];
+        yield ['1.525', '0.475', '0.724375'];
+        yield ['1.00', '1.000', '1.00000'];
+    }
+
+    public function testDivideIsImmutable(): void
+    {
+        $money = Money::fromString('1');
+
+        $money->divide('2', 1);
+
+        $this->assertEquals('1', $money->toString());
+    }
+
+    /**
+     * @dataProvider divideProvider
+     * @param string $a
+     * @param string $b
+     * @param int $scale
+     * @param string $expectedResult
+     */
+    public function testDivide(string $a, string $b, int $scale, string $expectedResult): void
+    {
+        $moneyA = Money::fromString($a);
+
+        $result = $moneyA->divide($b, $scale);
+
+        $this->assertEquals($expectedResult, $result->toString());
```

Comment:
Please use `assertSame` in all tests

Reply:
We have used this in all cases except one (`testRoundingIsConsistentWithPhpRound` formerly `testRoundingIsSameAsInPhp`)